File: lib/models/appModels/EventUserParticipant.dart
import 'EVent.dart';
import 'User.dart';
import 'Organization.dart';

class EventUserParticipant {
  Event? event;
  User? user;
  String? roles;
  Organization? organization;
  EventUserParticipantAttendingOptions? isAttending;

  EventUserParticipant({
    this.event,
    this.user,
    this.roles,
    this.organization,
    this.isAttending,
  });
}

File: apollo/src/EventUserParticipant.ts
import mongoose from 'mongoose';
import {EventUserParticipantAttendingOptions} from './enums.js'; 

const Schema = mongoose.Schema;


const eventUserParticipantSchema = new mongoose.Schema({
    event: { type: Schema.Types.ObjectId, ref: 'Event' },
    user: { type: Schema.Types.ObjectId, ref: 'User' },
    roles: String,    
    // isAttending: Object.values(EventUserParticipantAttendingOptions)
});

const EventUserParticipant = mongoose.model('EventUserParticipant', eventUserParticipantSchema);

export default EventUserParticipant;